apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerClientCreationOptions.cs

```diff
@@ -120,6 +135,19 @@ public override string ToString()
         /// </summary>
         internal async Task<ChannelCredentials> GetCredentialsAsync()
         {
+            // Create a builder that may create a client that connects to the
+            // emulator (if configured) so that we can extract the channel
+            // credentials used to connect to the emulator.
+            var builder = new SpannerClientBuilder
+            {
+                EmulatorDetection = EmulatorDetection
+            };
+            var emulatorBuilder = builder.MaybeCreateEmulatorClientBuilder();
+            if (emulatorBuilder != null)
+            {
+                return emulatorBuilder.ChannelCredentials;
+            }
+
```

Comment:
I was a bit surprised I had to do this.

I was attempting to call this function conditionally in `SessionPoolManager.CreateClientAsync()` only if `MaybeCreateEmulatorClientBuilder()` returns null. But this call kept happening and returning an error saying that a credentials files is not given  so I assume that it has something to do with the fact that it's async. I was trying to read up on when async functions get called but didn't find a good source that explained my problem.

Reply:
I can't tell what the code looked like when it was failing, but it seems reasonable to me. I don't think this has anything to do with async.

Basically if we want to use the emulator, we need to specify insecure channel credentials - otherwise, we use the normal code.

I'd probably write it in one step though:

```csharp
var emulatorBuilder = new SpannerClientBuilder { EmulatorDetection = EmulatorDetection }.MaybeCreateEmulatorClientBuilder();
...
```

If we use it elsewhere for the endpoint, you might want to put this into a common method.

Reply:
Thanks for the suggestion. Now it's in a separate function and I've cleaned up the calls a bit. Now there's no need to modify SessionPoolManager at all and all the changes are confined to this file.